mesoshttp/client.py

```diff
@@ -366,10 +386,32 @@ def set_credentials(self, principal, secret):
         :param secret: password
         :type secret: str
         '''
-        self.authenticate = True
         self.principal = principal
         self.secret = secret
 
+    def set_service_account(self, service_secret, verify=False):
+        '''
+        Set credentials to authenticate with DCOS and Mesos Master
+
+        :param service_secret: Optional DCOS Service account secret. Supersedes principal / secret.
+        :type service_secret: dict
+        :param verify: validate HTTPS fronted Mesos API using CA root trusts, defaults to False
+        :type verify: bool
+        '''
+
+        self.requests_auth = DCOSServiceAuth(service_secret)
+        self.principal = self.requests_auth.principal
+
+        cert_file = 'dcos-ca.crt'
+
+        if not verify:
+            response = requests.get('https://leader.mesos/ca/' + cert_file, verify=False)
```

Comment:
this is an hard coded uri "leader.mesos", is this mesos dns related name known by framework ? Don't think so it framework is not a DCOS service itself

Reply:
In DCOS, all tasks running under Mesos have access to the `leader.mesos` address as a means to route to the elected Mesos master. The `set_service_account` function would only be used under DCOS EE, as service accounts aren't even available in DCOS Community (and obviously in pure Mesos deployments), so this seemed like an acceptable approach. 

I included the inline attempt to pull the CA public key up front, only to limit the number of unverified TLS requests to 1. This seemed like the best effort in the case a trust bundle is provided by the package consumer. The bundle can be provided either as a parameter on `MesosClient` construction or by requests environment variable `REQUESTS_CA_BUNDLE`.

I'm open to other suggestions here.

Reply:
ok, will check further, but if it makes sense only in DCOS EE and as such makes use of mesos dns, I'm ok with that